Marlin/planner.h

```diff
@@ -52,7 +52,7 @@
 // the source g-code and may never actually be reached if acceleration management is active.
 typedef struct {
   // Fields used by the bresenham algorithm for tracing the line
-  long steps[NUM_AXIS];                     // Step count along each axis
+  unsigned long steps[NUM_AXIS];            // Step count along each axis
   unsigned long step_event_count;           // The number of step events required to complete this block
```

Comment:
I see that `steps` and `step_event_count` are treated as though they cannot go over `0x7FFFFFFF` in the Bresenham algorithm. The negative of the value is assigned to a counter. Then the algorithm looks for the counter to go over zero. If `steps` or `step_event_count` are ever `0x80000000` or higher, the code will fail. But surely there could never be more than 2.14 billion steps in a single move, right?